tool/src/java/mod.rs

```diff
@@ -0,0 +1,104 @@
+use std::collections::HashSet;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+use diplomat_core::hir::TypeContext;
+
+use crate::c2;
+use crate::common::FileMap;
+
+const TMP_C_DIR: &str = "tmp";
+const LIBRARY: &str = "somelib"; // todo: build from conf. Ensure that name is not the same as any
+                                 // type
+const GROUP: &str = "dev.diplomattest"; // todo: config
+const TMP_LIB_NAME: &str = "dev/diplomattest/somelib"; // todo: build from conf
+const JAVA_DIR: &str = "src/main/java/";
+
+pub fn run(
+    tcx: &TypeContext,
+    _conf_path: Option<&Path>,
+    out_folder: &Path,
+) -> std::io::Result<FileMap> {
+    let files = FileMap::default();
+    let mut context = c2::CContext::new(tcx, files, false);
+    context.run();
+
+    let errors = context.errors.take_all();
+
+    if !errors.is_empty() {
+        eprintln!("Found errors when generating c  code");
+        for error in errors {
+            eprintln!("\t{}: {}", error.0, error.1);
+        }
+    }
+
+    let out_files = context.files.take_files();
+
+    let tmp_path = out_folder.join(TMP_C_DIR);
+    std::fs::create_dir(&tmp_path)?;
+    let mut include_files = HashSet::new();
+    for (subpath, text) in out_files {
+        let out_path = tmp_path.join(&subpath);
+        if !subpath.ends_with(".d.h") && subpath.ends_with(".h") {
+            include_files.insert(subpath);
+        }
+        let parent = out_path
+            .parent()
+            .expect("Cannot create files at top level dir /");
+        std::fs::create_dir_all(parent)?;
+        let mut out_file = File::create(&out_path)?;
+        out_file.write_all(text.as_bytes())?;
+    }
+
+    let lib_path = tmp_path.join(format!("{LIBRARY}.h"));
+
+    let mut lib_file = File::create(&lib_path)?;
+    for include in include_files {
+        writeln!(lib_file, "#include \"{include}\"")?;
```

Comment:
jextract suggests loading all headers into a single library file. This gets transformed into an enormous monster of java